packages/babel-plugin-transform-es2015-parameters/test/fixtures/parameters/regression-5787/expected.js

```diff
@@ -0,0 +1,8 @@
+function f(a) {
+  var _ref, _ref2, _ref3;
+
+  var b = (_ref = (arguments.length <= 1 ? 0 : arguments.length - 1) - 3 + 1, arguments.length <= _ref || _ref < 1 ? undefined : arguments[_ref]);
```

Comment:
Good point. However, that check is generated by length-getter optimizer and necessary when `rest.length` is used in other places that is not as part of the index-getter optimizer. If we want to fix this, we have to keep track of whether we run the length-getter inside the context of an index-getter or not to generate the right code. I'm concerned about the added complexity to remove that test, not sure if it's worth it. But I can certainly give it a try.